enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+
+    def my_each
+        for i in 0..self.length-1
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0..self.length-1
+            yield(self[i],i)
+        end
+    end
+
+    def my_select
+        arr = []
+        for i in 1..self.length
+            if yield(i)
+                arr.push(i)
+            end
+        end
+        arr
+    end
+
+    def my_all?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_any?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_none?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = false
+            end
+        end
+        results
+    end
+
+    def my_count
+        counter = 0
+        for i in 0..self.length-1
+            if yield(self[i])
+                counter +=1
+            end
+        end
+        counter
+    end
+
+    def my_map(proc= nil)
+        array = []
+        for i in 1..self.length
+            yield(self[i])
+            array.push(i)
+        end
+        array
+    end
+    
+    def my_inject(total = nil)
+        total = self[0] unless total
+        for a in 1..self.length - 1
+            num1 = self[a]
+            total = yield total,num1
+        end
+        total
+    end
```

Comment:
* [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```ruby
    range = Range.new(5, 50)
    p range.my_inject(4) { |prod, n| prod * n } # returns this error
    # NoMethodError:
       #  undefined method `length' for 5..50:Range
      # ./enumerables.rb:76:in `my_inject'
  ```
* [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```ruby
    array = [1,7,8,89,9,6,5,12,4,2,5]
    p array.my_inject(:+) == array.inject(:+) # return this error
    # LocalJumpError:
       # no block given (yield)
      # ./enumerables.rb:78:in `block in my_inject'
      # ./enumerables.rb:76:in `each'
      # ./enumerables.rb:76:in `my_inject'
  ```